### PR TITLE
calligra: 3.2.1 -> 3.2.1-unstable-2023-05-28

### LINKED ISSUE #232868: Update request: calligra 3.2.1 → 3.3.0
- Package name: calligra
- Latest released version: 3.3.0

- Current version on the unstable channel: 3.2.1
- Current version on the stable/release channel: 3.2.1

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**

@ebzzry 
@zraexy 

* https://calligra.org/news/announcements/stable/calligra-plan-version-3-3-0-released/index.html
* https://download.kde.org/stable/calligra/3.3.0/

-----

Note for maintainers: Please tag this issue in your PR.

### PR DESCRIPTION
**Pull request by MGlolenstine**
###### Description of changes
Closes #232868
Changed the source provider to GitHub because I couldn't find the `tar.xz` sources for the Calligra suite.
Removed two patches, as they failed to apply and weren't necessary for the build to succeed.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/office/calligra/default.nix [hunk 1]
```diff
@@ -14,28 +14,17 @@
...
-  src = fetchurl {
-    url = "mirror://kde/stable/${pname}/${version}/${pname}-${version}.tar.xz";
-    sha256 = "0iqi6z6gkck2afgy200dacgcspq7i7887alcj0pklm08hbmsdy5i";
+  src = fetchFromGitHub {
+    owner = "KDE";
+    repo = "calligra";
+    # Last commit from "calligra/3.3" branch as of now
+    rev = "2301b2c49f7ec99214a42bdde0b3aadaf05adc6a";
+    sha256 = "sha256-gQUSL7k34HNZ3O65RO+wb+C4MEcq1U577p40QDyGPKs=";
   };
```
Comment (wegank) [inline pkgs/applications/office/calligra/default.nix:25]: Please fetch from https://invent.kde.org/office/calligra, thanks.

File: pkgs/applications/office/calligra/default.nix [hunk 2]
```diff
@@ -14,28 +14,17 @@
 
 mkDerivation rec {
   pname = "calligra";
-  version = "3.2.1";
+  version = "3.3.0";
```
Comment (wegank) [inline pkgs/applications/office/calligra/default.nix:17]:
```suggestion
  version = "unstable-2023-05-28";
```

File: pkgs/applications/office/calligra/default.nix [hunk 3]
```diff
@@ -14,28 +14,15 @@
 
 mkDerivation rec {
   pname = "calligra";
-  version = "3.2.1";
+  version = "unstable-2023-05-28";
 
-  src = fetchurl {
-    url = "mirror://kde/stable/${pname}/${version}/${pname}-${version}.tar.xz";
-    sha256 = "0iqi6z6gkck2afgy200dacgcspq7i7887alcj0pklm08hbmsdy5i";
+  src = fetchGit {
```
Comment (wegank) [inline pkgs/applications/office/calligra/default.nix:19]: `fetchFromGitLab` should be used here.

**Comment by Aleksanaa** [comment]
bump

**Comment by SigmaSquadron** [comment]
Superseded by #344930

**Pull request closed without merging**